### ISSUE TITLE
CHANNEL_LOGO_EXTENSION not appended to logo path

### ISSUE THREAD

**Issue by jaylinski** [kodi-pvr/pvr.iptvsimple#156]
The documentation for IPTV Simple Client states the following:

> tvg-logo is name of channel logo file **without extension (.png)**. If this tag is absent then addon will use channel name to find logo.

http://kodi.wiki/view/Add-on:IPTV_Simple_Client

This seems to be untrue, because the `CHANNEL_LOGO_EXTENSION` constant is never appended to the logo path:

https://github.com/kodi-pvr/pvr.iptvsimple/blob/00f19965d03db9b6cf7fbef98ed34c87b0d6bf4b/src/PVRIptvData.cpp#L1025

There are two possible solutions: 

1. Fix the code by appending the logo extension if there is no extension present in the `tvg-logo` string (backwards compatible in order not to break existing playlists)
2. Update the docs to `tvg-logo is name of channel logo file with the extension ".png" ...` and remove the unused constant from the code

Which one do you prefer?

**Comment by allixx** [comment]
Missing ".png" append is not a problem for local filesystem logo source (some loader class does its magic under the hood I guess). 

When logo source is Remote Path (Internet address) and strTvgLogo is autogenerated from channel name, ".png" appending must definitely happen.

**Comment by jaylinski** [comment]
@allixx Thanks for your input! I already wondered why nobody else seems to have this problem. You guessed right, I'm using remote paths for the icons.

```xml
<channel id="14">
  <display-name lang="de">ORFeins</display-name>
  <display-name lang="de">ORFeins</display-name>
  <icon src="http://[redacted].net/img/station/darkbg/200x200/14.png" width="" height=""/>
</channel>
```
This behavior now makes sense to me. But it still think this should be mentioned in the docs.

**Comment by 32Dexter** [comment]
Hi everyone,
I have the same problem, I have a channel list in which I am not the administrator and in the list m3u the "tvg-logo" is empty (tvg-logo = "").
It must therefore resolve in another way, from what I understand, I am on this page:
https://kodilive.eu/tutorial/min/imp_logo.jpg
And there are 3 solutions:
1. Set in the "Base URL" the directory where all the logos with the same name of the channels are contained (but checking the log does not seem to be added to the ".png" extension)
http://www.arthasdesign.it/logos/
2. Set in the "Base URL" the XML file that contains the position of the logos, example:
http://www.arthasdesign.it/loghi_new/xmltv2.xml
3. Set the M3U file containing the position of the logos in the "Base URL", example:
http://www.arthasdesign.it/loghi_new/logos.m3u
Nothing is set on the EPG page yet.
Unfortunately the solution 2 and solution 3 I don't understand why it doesn't work ...
Can you help me?

**Comment by phunkyfish** [comment]
Fixed in #269 and #275
